tests/src/rules/no-duplicates.js

```diff
@@ -128,6 +128,22 @@ ruleTester.run('no-duplicates', rule, {
       errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
     }),
 
+    // #1403: first default and rest named imports should be merged
+    test({
+      code: "import def from './foo'; import {x} from './foo';",
+      output: "import def, {x} from './foo'; ",
+      parser: require.resolve('babel-eslint'),
+      errors: ['\'./foo\' imported multiple times.', '\'./foo\' imported multiple times.'],
+    }),
+
+    // #1403: first default and rest named imports should be merged
+    test({
+      code: "import def, {x} from './foo'; import {y} from './foo';",
```

Comment:
can we add some tests that include merging multiple default imports of the same name, and also like 3 or 4 different import statements bringing in names?

Reply:
> can we add some tests that include merging multiple default imports of the same name,

Multiple default imports from the same module will be bailed by https://github.com/benmosher/eslint-plugin-import/blob/master/src/rules/no-duplicates.js#L53

> and also like 3 or 4 different import statements bringing in names?

I didn't quite understand what you mean by that. I checked with 
```js
import { x, y } from './foo'; import { y, z } from './foo';
```
and was it fixed to 
```js
import { x, y , y, z } from './foo' 
```
And I think that it's right because developer should handle duplicated variables.

Reply:
Why? If they’re the same value and the same name and they’re already being merged, they should be uniquified.

Also, what about named imports that rename using `as`?

Reply:
Oh, I see your point now. 

```js
// before
import def from './foo'; import {default as x} from './foo';
// after
import def, {default as x} from './foo'; 

// before
import { x } from './foo'; import { x as y } from './foo';
// after
import { x , x as y } from './foo'; 
```

I just thought that this not fits in the scope of the current pull request.